ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java

```diff
@@ -0,0 +1,1240 @@
+          node.getSign(),
+          (Expression) process(node.getValue(), context)
+      );
+    } else {
+      return new ArithmeticUnaryExpression(
+          node.getSign(),
+          (Expression) process(node.getValue(), context)
+      );
+    }
+
+  }
+
+  protected Node visitNotExpression(NotExpression node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new NotExpression(node.getLocation().get(),
+          (Expression) process(node.getValue(), context)
+      );
+    } else {
+      return new NotExpression((Expression) process(node.getValue(), context)
+      );
+    }
+
+  }
+
+  protected Node visitSelectItem(SelectItem node, Object context) {
+    return visitNode(node, context);
+  }
+
+  protected Node visitSingleColumn(SingleColumn node, Object context) {
+
+    if (node.getLocation().isPresent()) {
+      return new SingleColumn(node.getLocation().get(),
+          (Expression) process(node.getExpression(), context),
+          node.getAlias()
+      );
+    } else {
+      return new SingleColumn(
+          (Expression) process(node.getExpression(), context),
+          node.getAlias()
+      );
+    }
+
+  }
+
+  protected Node visitAllColumns(AllColumns node, Object context) {
+    if (node.getLocation().isPresent()) {
+      if (node.getPrefix().isPresent()) {
+        return new AllColumns(node.getLocation().get(), node.getPrefix().get());
+      } else {
+        return new AllColumns(node.getLocation().get());
+      }
+
+    } else {
+      if (node.getPrefix().isPresent()) {
+        return new AllColumns(node.getLocation().get());
+      } else {
+        throw new KsqlException("Cannot have both localtion and prefix null in AllColumns AST "
+            + "node.");
+      }
+
+    }
+
+  }
+
+  protected Node visitSearchedCaseExpression(SearchedCaseExpression node, Object context) {
+    Optional<Expression> defaultValue = node.getDefaultValue().isPresent()
+        ? Optional.ofNullable((Expression) process(node.getDefaultValue().get(), context))
+        : Optional.empty();
+    if (node.getLocation().isPresent()) {
+      return new SearchedCaseExpression(
+          node.getLocation().get(),
+          node.getWhenClauses()
+              .stream().map(whenClause -> (WhenClause) process(whenClause, context))
+              .collect(Collectors.toList()),
+          defaultValue
+      );
+    } else {
+      return new SearchedCaseExpression(
+          node.getWhenClauses()
+              .stream().map(whenClause -> (WhenClause) process(whenClause, context))
+              .collect(Collectors.toList()),
+          defaultValue
+      );
+    }
+
+  }
+
+  protected Node visitLikePredicate(LikePredicate node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new LikePredicate(node.getLocation().get(),
+          (Expression) process(node.getValue(), context),
+          (Expression) process(node.getPattern(), context),
+          node.getEscape() != null
```

Comment:
nit: I know its not in the purview of this diff but would be good to make the return value here Optional

Reply:
Yes, but that would need changes in parent classes too. We can do this in a separate PR.